graphjet-core/src/main/java/com/twitter/graphjet/algorithms/ExactUserSocialProofSizeFilter.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2017 Twitter. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twitter.graphjet.algorithms;
+
+import com.twitter.graphjet.hashing.SmallArrayBasedLongToDoubleMap;
+import com.twitter.graphjet.stats.StatsReceiver;
+
+/**
+ * Only allow nodes whose total number of user social proofs is exactly equal to
```

Comment:
consistent comment style with other concrete filters? for example
This filter keeps the nodes with the exact number of user social proofs equal to the parameter of the query, no more, no less.